circuit-lib/circuit-lib.js/tests/elgamal-babyjubjub/elgamal.test.ts

```diff
@@ -0,0 +1,274 @@
+import { assert, expect } from "chai";
+import {
+  babyJub,
+  genKeypair,
+  decrypt,
+  rerandomize,
+} from "../../src/elgamal-babyjubjub/elgamal";
+
+import { getRandomBytes, encrypt_demo, encrypt_encoded } from "./test-utils";
+
+import {
+  encode,
+  decode,
+  split64,
+  precompute,
+} from "../../src/elgamal-babyjubjub/decode";
+
+const fs = require("fs");
+
+const TWO_32 = 4294967296n;
+
+describe("Testing ElGamal Scheme on EC points directly", () => {
+  it("Check compliance of orignal and decrypted message as points", () => {
+    const keypair = genKeypair();
+    const encryption = encrypt_encoded(keypair.pubKey);
```

Comment:
- The `encrypt` function isn't indeed used directly in test because:
    - Based on security it shouldn't leak the message or nonce.
    - the nonce or message encoding could be done out of the function but that's not practical for the user.
    - I created test versions of it to test homomorphism and other cases
    - At the end, i didn't use it in tests but I used unsecure copy of it.